includes/handlers/class-webmention-handler-meta.php

```diff
@@ -75,150 +37,29 @@ private function parse_meta( $dom ) {
-					} elseif ( isset( $meta['article']['modified'] ) ) {
-						$this->webmention_item->set_updated( new DateTimeImmutable( $meta['article']['modified'] ) );
-					}
-				}
-			}
-		}
-	}
-
-	protected function dublin_core_to_jf2( $meta ) {
-		$item = array();
-		// Then look at Dublin Core Properties
-		if ( isset( $meta['dc'] ) ) {
-			$dc = $meta['dc'];
-			if ( isset( $dc['Title'] ) ) {
-				$item['name'] = $dc['Title'];
-			}
-			if ( isset( $dc['Creator'] ) ) {
-				if ( is_string( $dc['Creator'] ) ) {
-					$this->webmention_item->set_author( array( 'name' => $dc['Creator'] ) );
-				}
-			}
-			if ( isset( $dc['Description'] ) ) {
-				$this->webmention_item->set_summary( $dc['Description'] );
-			}
-			if ( isset( $dc['Date'] ) ) {
-				$this->webmention_item->set_published( new DateTimeImmutable( $dc['Date'] ) );
-			}
-		}
-	}
+		// OGP has no concept of anything but mention so it is always a mention.
+		$this->webmention_item->set__response_type = 'mention';
 
-	protected function add_property( $array, $key, $value ) {
-		if ( ! isset( $array[ $key ] ) ) {
-			$array[ $key ] = $value;
-		} elseif ( is_string( $array[ $key ] ) ) {
-			$array[ $key ] = array( $array[ $key ], $value );
-		} elseif ( is_array( $array[ $key ] ) ) {
-			$array[ $key ][] = $value;
+		// If Site Name is not set use domain name less www
+		if ( ! $this->webmention_item->has_site_name() && $this->webmention_item->has_url() ) {
+			$this->webmention_item->set__site_name( preg_replace( '/^www\./', '', wp_parse_url( $this->webmention_item->get_url(), PHP_URL_HOST ) ) );
 		}
-		return $array;
 	}
 
-	/**
-	 * Tries to parse OGP Meta tags by hierarchy.
-	 */
-	protected function parse_ogp( $meta ) {
-		$return = array();
-		if ( isset( $meta ) && is_array( $meta ) ) {
-			foreach ( $meta as $key => $value ) {
-				$nmeame = explode( ':', $key );
-				if ( 1 === count( $name ) ) {
-					$name = explode( '.', $key );
-				}
-				if ( 1 < count( $name ) ) {
-					$name = $name[0];
-					$key  = str_replace( $name . ':', '', $key );
-					$key  = str_replace( $name . '.', '', $key );
-					if ( is_array( $value ) ) {
-						$value = array_unique( $value );
-						if ( 1 === count( $value ) ) {
-							$value = array_shift( $value );
-						}
-					}
-					if ( ! isset( $return[ $name ] ) ) {
-						$return[ $name ] = array(
-							$key => $value,
-						);
-					} else {
-						if ( is_string( $return[ $name ] ) ) {
-							$return[ $name ] = array( $return[ $name ] );
-						}
-						$return[ $name ][ $key ] = $value;
-					}
-				} else {
-					$return[ $key ] = $value;
-				}
-			}
+	protected function add_property( $key, $value ) {
+		if ( in_array( $key, array( 'url', 'og:url' ), true ) ) {
+			$this->webmention_item->set_url( $value );
+		} elseif ( in_array( $key, array( 'DC.Title', 'og:title' ), true ) ) {
+			$this->webmention_item->set_name( $value );
 		}
```

Comment:
this is only exemplary and I should re-add the missing ones

Reply:
Yes, it does look much simpler.